sunkit_instruments/rhessi/rhessi.py

```diff
@@ -350,3 +351,53 @@ def _build_energy_bands(label, bands):
     unit = matched.group('UNIT').strip()
 
     return [f'{band} {unit}' for band in bands]
+
+
+def hsi_fits2map(image_datacube):
+    """
+    Extracts single map images from a RHESSI flare image datacube.
+    Currently assumes input to be 4D.
+
+    Parameters
+    ----------
+    image_datacube: `str`
+        Path or URL to image datacube .fits
+
+    Returns
+    -------
+    `dict` of `list` of `sunpy.map.Map`
+        Each energy band has a list of maps where the index of the lists represent the time step
+    """
+    # import sunpy.map in here so that net and timeseries don't end up importing map
+    from sunpy.map import Map
+    
+    f = sunpy.io.read_file(image_datacube)
+    header = f[0].header
+    
+    # remove those (non-standard) headers to avoid user warnings (they are 0 anyway)
+    del header["CROTACN1"]
+    del header["CROTACN2"]
+    del header["CROTA"]
+
+    d_min = {}
+    d_max = {}
+    for e in range(len(f[1].data[0]["ENERGY_AXIS"])):
```

Comment:
I think there is issues here for when there is an image cube of a time sequence at one energy range. If there is only one energy range, say for example 12-25keV, the `len(f[1].data[0]["energy_axis"]` here is going to to be  `array([12., 25.], dtype=float32)` of length 2 so will have issues indexing below as the `f[0].data.shape is (x, 1, x, x)` So there should be a check here for those cases